src/components/ExportGithub.tsx

```diff
@@ -0,0 +1,211 @@
+import { Button } from "@blueprintjs/core";
+import React, { useEffect, useRef, useState } from "react";
+import MenuItemSelect from "roamjs-components/components/MenuItemSelect";
+import apiGet from "roamjs-components/util/apiGet";
+import localStorageSet from "roamjs-components/util/localStorageSet";
+
+type UserReposResponse = {
+  data: [
+    {
+      name: string;
+      full_name: string;
+    }
+  ];
+  status: number;
+};
+type UserRepos = UserReposResponse["data"];
+const initialRepos: UserRepos = [{ name: "", full_name: "" }];
+
+const WINDOW_WIDTH = 600;
+const WINDOW_HEIGHT = 525;
+const WINDOW_LEFT = window.screenX + (window.innerWidth - WINDOW_WIDTH) / 2;
+const WINDOW_TOP = window.screenY + (window.innerHeight - WINDOW_HEIGHT) / 2;
+
+export const ExportGithub = ({
+  isVisible,
+  selectedRepo,
+  setSelectedRepo,
+  setError,
+  clearGitHubToken,
+  gitHubAccessToken,
+  setGitHubAccessToken,
+  setCanSendToGitHub,
+}: {
+  isVisible: boolean;
+  selectedRepo: string;
+  setSelectedRepo: (selectedRepo: string) => void;
+  setError: (error: string) => void;
+  clearGitHubToken: boolean;
+  gitHubAccessToken: string | null;
+  setGitHubAccessToken: (gitHubAccessToken: string | null) => void;
+  setCanSendToGitHub: (canSendToGitHub: boolean) => void;
+}) => {
+  const authWindow = useRef<Window | null>(null);
+
+  const [isGitHubAppInstalled, setIsGitHubAppInstalled] = useState(false);
+  const [clickedInstall, setClickedInstall] = useState(false);
+  const [repos, setRepos] = useState<UserRepos>(initialRepos);
+  const showGitHubLogin = isGitHubAppInstalled && !gitHubAccessToken;
+  const repoSelectEnabled = isGitHubAppInstalled && gitHubAccessToken;
+
+  const setRepo = (repo: string) => {
+    setSelectedRepo(repo);
+    localStorageSet("selected-repo", repo);
+  };
+
+  const clearAuth = () => {
+    setGitHubAccessToken(null);
+    localStorageSet("oauth-github", "");
+  };
+
+  const fetchAndSetInstallation = async () => {
+    try {
+      const res = await apiGet<{ installations: { app_id: number }[] }>({
+        domain: "https://api.github.com",
+        path: "user/installations",
+        headers: {
+          Authorization: `token ${gitHubAccessToken}`,
```

Comment:
[nit] I would access `gitHubAccessToken` from local storage which allows us to wrap this function as a useCallback